AG-16452 - Make benchmark UI less intrusive with floating overlay panel

alantreadway opened the pull request:
https://ag-grid.atlassian.net/browse/AG-16452

alantreadway committed:
Make benchmark UI less intrusive with floating overlay panel

- Merge status bar and export button into collapsible bottom panel
- Floating run button when no example-controls exist
- Semi-transparent overlay backgrounds (0.8 opacity)
- Collapsible panel with toggle button always visible at bottom edge
- Progress bar and results table now float over example content

alantreadway committed:
Preserve original example-controls when running benchmarks

Only hide the run button itself, not the entire controls row, so that
original example controls remain visible during benchmark execution.

lsjroberts commented on plugins/ag-charts-generate-example-files/src/executors/generate/generator/utils/benchmarkHarness.src.ts:
```diff
@@ -301,10 +327,44 @@ class BenchmarkUI {
+            if (computedStyle.position === 'static') {
+                chartParent.style.position = 'relative';
+            }
+        }
```
Duplicated in `createBenchmarkContainer()`, maybe extract out.

alantreadway replied:
Done, thanks

lsjroberts reviewed (approved):
LGTM

alantreadway committed:
Extract chart parent positioning logic into helper method

Addresses PR feedback to reduce duplication between createBenchmarkContainer()
and createFloatingButton().

alantreadway committed:
Rename benchmarkHarness.src.ts to benchmarkHarness.ts

Remove redundant .src suffix from filename and update references
in benchmarkRunner.ts.
